### PR TITLE
Combine PasswordInput and PasswordInputStateful

### PR DESCRIPTION
**Pull request by conor-cafferkey-sociomantic**
This PR:
- Combines the  PasswordInput and PasswordInputStateful components into a single PasswordInput component

### CODE REVIEW

File: src/defaults.json [hunk 1]
```diff
@@ -627,11 +627,7 @@
 			}
 		]
 	},
-	"PasswordInput": {
-		"label": "Tell me your secrets",
-		"errorMessage": "Please enter a valid name"
-	},
-	"PasswordInputStateful": {
+	"PasswordInpu": {
```
Comment (mahmoud-zakria-sociomantic) [inline src/defaults.json]: Missing `t`

Comment (daniel-martic-sociomantic) [inline src/defaults.json]:
```suggestion
	"PasswordInput": {
```
Mr. T was not invited ☹️

File: src/PasswordInput/index.jsx [hunk 2]
```diff
@@ -10,231 +10,236 @@
 import React                 from 'react';
 import PropTypes             from 'prop-types';
 
+import { TextInputWithIcon } from '../';
```
Comment (daniel-martic-sociomantic) [inline src/PasswordInput/index.jsx]: Although this works, linter is complaining `Dependency cycle detected (import/no-cycle)`, even if we leave `'../index'`. Can you please check if this is ok, and should we do something about this rule?

File: src/defaults.json (hunk 1 above)
Comment (conor-cafferkey-sociomantic) [reply src/defaults.json]: 😄

File: src/PasswordInput/index.jsx (hunk 2 above)
Comment (conor-cafferkey-sociomantic) [reply src/PasswordInput/index.jsx]: Yeah I did a `yarn upgrade --latest` in the _Nessie-X-alpha0_ branch and it introduced a bunch of weird new rules. Will fix it in the main branch.

**Comment by mahmoud-zakria-sociomantic** [comment]
LGTM 💥

**Pull request merged**